--- test.urdf	2014-01-18 20:01:44.102247274 -0600
+++ /tmp/test.urdf	2014-01-18 20:01:50.822247507 -0600
@@ -1,8 +1,5 @@
 <?xml version="1.0"?>
 <robot name="test">
-  <!-- Examples from: http://wiki.ros.org/urdf/XML/Transmission -->
-
-  <!-- New Transmission -->
   <transmission name="simple_trans">
     <type>transmission_interface/SimpleTransmission</type>
     <joint name="foo_joint"/>
@@ -11,12 +8,9 @@
       <mechanicalReduction>50.0</mechanicalReduction>
     </actuator>
   </transmission>
-
-  <!-- PR2 Transmission -->
   <transmission name="caster_front_left_trans" type="SimpleTransmission">
     <joint name="caster_front_left_joint"/>
     <actuator name="caster_front_left_motor"/>
     <mechanicalReduction>1.0</mechanicalReduction>
   </transmission>
-
 </robot>
