--- /tmp/pelican.urdf	2013-04-11 09:24:30.325804322 -0500
+++ /tmp/tmp.urdf	2013-04-11 09:26:50.185809508 -0500
@@ -1,111 +1,78 @@
+<?xml version="1.0"?>
 <robot name="Asctec Pelican">
-  
-################################################################################
-# LINKS                                                                        #
-################################################################################
-
-  ##### base_link ##############################################################
-
   <link name="base_link">
     <visual>
-      <origin xyz="0.0 0.0 0.0" rpy="1.57 0.0 3.1415"/>
+      <origin rpy="1.57 0.0 3.1415" xyz="0.0 0.0 0.0"/>
       <geometry>
-        <mesh filename="package://pelican_urdf/data/pelican.stl" 
-              scale="0.001 0.001 0.001"/>
+        <mesh filename="package://pelican_urdf/data/pelican.stl" scale="0.001 0.001 0.001"/>
       </geometry>
       <material name="blue">
         <color rgba="0.4 0.4 0.5 1.0"/>
       </material>
     </visual>
   </link>
-
-  ##### base_footprint #########################################################
-
   <link name="base_footprint"/>
-
-  ##### legs ###################################################################
-
   <link name="leg_1">
     <visual>
+      <origin rpy="0.0 0.0 0.0" xyz="0.089 0.0 -0.057"/>
       <geometry>
-       <box size="0.044 0.007 0.114"/>
+        <box size="0.044 0.007 0.114"/>
       </geometry>
-      <origin xyz="0.089 0.00 -0.057" rpy="0.0 0.0 0.0"/>
       <material name="g">
         <color rgba="0.2 0.25 0.25 1.0"/>
       </material>
     </visual>
   </link>
-
   <link name="leg_2">
     <visual>
+      <origin rpy="0.0 0.0 0.0" xyz="-0.089 0.0 -0.057"/>
       <geometry>
-       <box size="0.044 0.007 0.114"/>
+        <box size="0.044 0.007 0.114"/>
       </geometry>
-      <origin xyz="-0.089 0.00 -0.057" rpy="0.0 0.0 0.0"/>
       <material name="g"/>
     </visual>
   </link>
-
   <link name="leg_3">
     <visual>
+      <origin rpy="0.0 0.0 1.5708" xyz="0.0 0.089 -0.057"/>
       <geometry>
-       <box size="0.044 0.007 0.114"/>
+        <box size="0.044 0.007 0.114"/>
       </geometry>
-      <origin xyz="0.00 0.089 -0.057" rpy="0.0 0.0 1.5708"/>
       <material name="g"/>
     </visual>
   </link>
-
   <link name="leg_4">
     <visual>
+      <origin rpy="0.0 0.0 1.5708" xyz="0.0 -0.089 -0.057"/>
       <geometry>
-       <box size="0.044 0.007 0.114"/>
+        <box size="0.044 0.007 0.114"/>
       </geometry>
-      <origin xyz="0.00 -0.089 -0.057" rpy="0.0 0.0 1.5708"/>
       <material name="g"/>
     </visual>
   </link>
-
-################################################################################
-# JOINTS                                                                       #
-################################################################################
-
-  ##### base_footprint #########################################################
-
   <joint name="base_link_to_base_footprint" type="fixed">
     <parent link="base_link"/>
     <child link="base_footprint"/>
-    <origin xyz="0.00 0.00 -0.195" rpy="0.0 0.0 0.0"/>
+    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.195"/>
   </joint>
-
-  ##### legs ###################################################################
-  
   <joint name="base_link_to_leg_1" type="fixed">
     <parent link="base_link"/>
     <child link="leg_1"/>
-    <origin xyz="0.00 0.00 -0.083" rpy="0.0 0.0 0.0"/>
+    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.083"/>
   </joint>
-
   <joint name="base_link_to_leg_2" type="fixed">
     <parent link="base_link"/>
     <child link="leg_2"/>
-    <origin xyz="0.00 0.00 -0.083" rpy="0.0 0.0 0.0"/>
+    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.083"/>
   </joint>
-
   <joint name="base_link_to_leg_3" type="fixed">
     <parent link="base_link"/>
     <child link="leg_3"/>
-    <origin xyz="0.00 0.00 -0.083" rpy="0.0 0.0 0.0"/>
+    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.083"/>
   </joint>
-
   <joint name="base_link_to_leg_4" type="fixed">
     <parent link="base_link"/>
     <child link="leg_4"/>
-    <origin xyz="0.00 0.00 -0.083" rpy="0.0 0.0 0.0"/>
+    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.083"/>
   </joint>
-
 </robot>
-
-
-
