Index: dom/Makefile
===================================================================
--- dom/Makefile	2008-10-02 17:48:30.000000000 -0700
+++ dom/Makefile	2010-03-03 14:44:09.162398249 -0800
@@ -55,7 +55,7 @@
 conf := release
 
 # Collada version: '1.4', '1.5', or 'all'
-colladaVersion := 1.4
+colladaVersion := 1.5
 
 # parser: 'libxml', 'tinyxml', or 'all'.
 parser := libxml
@@ -188,7 +188,7 @@
 ifneq ($(filter install uninstall installTest,$(MAKECMDGOALS)),)
 # You can only install on Mac or Linux. Check for that.
 ifeq ($(oss),linux)
-prefix := /usr/local
+prefix := ../../../
 else ifeq ($(oss),mac)
 prefix := /Library/Frameworks
 else
